### DISCUSSION TITLE
may sound stupid or trying to be funny but i was trying to test out in raspberry pi?

### DISCUSSION THREAD

**Discussion by kolinfluence** [cloudflare/workerd#219] [Q&A] [answered]
clang compilation seemed taking a long time.

out of curiosity, 
1. what is workerd platform being powered in / designed for?
2. i'm trying to run on raspberry pi 4 . advisable?
3. how many req/s for 
    a) http and
    b) connections per second for websocket
    can something like raspberry pi 4 or normal laptop run in?
4. what's the normal "memory used" per worker instance for this or how to minimize resources per worker instance "spawned"?
5. can you provide a roadmap / direction towards this repo? (to see suitability in ... future scenario usage)

**Comment by kentonv** [comment, marked as answer]
Hi @ultperf,

(I converted this to a discussion since it's a list of questions. We prefer to use issues strictly for actionable items requiring code changes.)

Yes, people have gotten `workerd` to work on a Raspberry Pi. It needs to be 64-bit. Yes, it'll take quite a while to compile, mostly because V8 is quite large. Alternatively, you should be able to use the precompiled aarch64 binary from npm. We do not know how it will perform on a Pi. You'll have to try it.

The `workerd` release has two puposes:

1. Support realistic local testing of code meant to run on Cloudflare Workers.
2. Enable code written for Cloudflare Workers to be hosted on any traditional Linux VM host, so you aren't "locked into" Cloudflare.

Our "roadmap" is primarily focused on building new features for Cloudflare Workers, while continuing to support the above two goals.
